Doc/library/tkinter.rst

```diff
@@ -67,16 +67,71 @@ Or, more often::
    from tkinter import *
 
 
-.. class:: Tk(screenName=None, baseName=None, className='Tk', useTk=1)
-
-   The :class:`Tk` class is instantiated without arguments. This creates a toplevel
-   widget of Tk which usually is the main window of an application. Each instance
-   has its own associated Tcl interpreter.
-
-   .. FIXME: The following keyword arguments are currently recognized:
-
-
-.. function:: Tcl(screenName=None, baseName=None, className='Tk', useTk=0)
+.. class:: Tk(screenName=None, baseName=None, className='Tk', useTk=True, sync=False, use=None)
+
+   Construct a toplevel Tk widget, which is usually the main window of an
+   application, and initialize a Tcl interpreter for this widget.  Each
+   instance has its own associated Tcl interpreter.
+
+   The :class:`Tk` class is typically instantiated using all default values.
+   However, the following keyword arguments are currently recognized:
+
+   *screenName*
+      When given (as a string), sets the :envvar:`DISPLAY` environment
+      variable. (X11 only)
+   *baseName*
+      Name of the profile file.  By default, *baseName* is derived from the
+      program name (``sys.argv[0]``).
+   *className*
+      Name of the widget class.  Used as a profile file and also as the name
+      with which Tcl is invoked (*argv0* in *interp*).
```

Comment:
So about baseName and className... 

My understanding is that className is the equivalent of wish's "-name" option. So this provides the default title for the root window, the class of the root window (which can be used in bindings and option database), and would be what is returned by "tk appname" (sometimes used for inter-application communication).

It appears that baseName isn't passed into Tcl/Tk at all.

About those profile files that can be read in... does anyone know what they're intended to be used for, if they've ever been documented, and if they're actually used at all? It reads like a per-application 'automatically inject your own code' which seems a bit strange.

Reply:
I tried out specifying className on a call and it seems to lower case the first letter:
``` python
root = tkinter.Tk(className='TEST')
```
creates a window with the name `tEST` and `root.tk.call('tk', 'appname')` returns `tEST`.  Should I document that behavior? Since _fileName_ in the `wish` documentation is used to find the script, I think it might be lowercased for that reason?  


Looking through the [documentation](https://www.tcl.tk/man/tcl8.7/TkCmd/tk.htm#M5), it looks like `interp` is used for inter-application communication, which is the value stored in `Tk.tk`.  
```
tk appname ?newName?
    If newName is not specified, this command returns the name of the application (the name
    that may be used in **send** commands to communicate with the application)
```
However, the send command in tkinter has:
``` python
    def send(self, interp, cmd, *args):
        """Send Tcl command CMD to different interpreter INTERP to be executed."""
        return self.tk.call(('send', interp, cmd) + args)
```

As far as the profile files, there was a [security patch](https://bugs.python.org/issue16248) in the past.

Reply:
I'm not sure why the first letter is lowercased - it's Tkinter that's doing it, though for what reason I can't guess. From the Tcl side it's perfectly ok with the first letter of argv0 capitalized. My instinct therefore is not to document it.

Note that the classname for the window will lowercase everything except for uppercasing the first letter, e.g. "Test" with your example. I think the first letter uppercase is used for user class names vs. builtin ones like 'button' etc.  I was playing around with different things using this:

    r = Tk(className='TEST')
    print(r.call('tk','appname'))
    print(r.wm_title())
    print(r.bindtags())
    print(sys.argv[0])

I'd suggest not documenting this conversion either. Anyone who needs to get the exact capitalization of the classname will know how to get it through introspection. I also don't imagine that 'send' would be used much by Tkinter apps (it was a useful thing a long while ago, not so much now).

Any suggestions how to proceed with regard to the readprofile() thing, i.e. what on earth it is for and if it should still be there? @serhiy-storchaka ?